pkgs/development/web/publii/default.nix

```
@@ -0,0 +1,109 @@
+{ stdenv
+, lib
+, fetchurl
+, autoPatchelfHook
+, makeShellWrapper
+, wrapGAppsHook
+, alsa-lib
+, at-spi2-atk
+, at-spi2-core
+, atk
+, cairo
+, cups
+, dbus
+, expat
+, glib
+, gtk3
+, libsecret
+, mesa
+, nss
+, pango
+, udev
+, xdg-utils
+, xorg
+}: let
+  intName = "Publii";
+  version = "0.42.0";
+
+in stdenv.mkDerivation {
+  pname = lib.strings.toLower intName;
+  inherit version;
+
+  src = fetchurl {
+    url = "https://getpublii.com/download/Publii-${version}.deb";
+    hash = "sha256-ohwoKhLT75gOC4f6R1tjhSygNw5lR7kbWBG6qpcP368=";
+  };
+
+  dontConfigure = true;
+  dontBuild = true;
+  dontWrapGApps = true;
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    makeShellWrapper
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    alsa-lib
+    at-spi2-atk
+    at-spi2-core
+    atk
+    cairo
+    cups
+    dbus
+    expat
+    glib
+    gtk3
+    libsecret
+    mesa
+    nss
+    pango
+    xorg.libX11
+    xorg.libxcb
+  ];
+
+  unpackPhase = ''
+    ar p $src data.tar.xz | tar xJ
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+
+    mv usr/share $out
+    substituteInPlace $out/share/applications/${intName}.desktop \
+      --replace 'Exec=/opt/Publii/Publii' 'Exec=${intName}'
+
+    mv opt $out
+
+    runHook postInstall
+  '';
+
+  preFixup = ''
+    # makeWrapper defaults to makeBinaryWrapper due to wrapGAppsHook
+    # but we need a shell wrapper specifically for xdg-fix. Make xdg-open overrideable at runtime.
+    # Electron is trying to open udev via dlopen()
+    # and for some reason that doesn't seem to be impacted from the rpath.
+    # Adding udev to LD_LIBRARY_PATH fixes that.
+    makeShellWrapper $out/opt/${intName}/${intName} $out/bin/${intName} \
```

```suggestion
    makeShellWrapper $out/opt/Publii/Publii $out/bin/Publii \
```
having the files in opt is not that nice but works